core/lib/node/src/resource/mod.rs

```diff
@@ -0,0 +1,35 @@
+use std::{any::Any, fmt};
+
+pub mod object_store;
+pub mod pools;
+
+/// A marker trait for anything that can be stored (and retrieved) as a resource.
+/// Requires `Clone` since the same resource may be requested by several tasks.
+pub trait Resource: 'static + Clone + std::any::Any {
+    /// Unique identifier of the resource.
+    /// Used to fetch the resource from the provider.
+    ///
+    /// It is recommended to name resources in form of `<scope>/<name>`, where `<scope>` is the name of the task
+    /// that will use this resource, or 'common' in case it is used by several tasks, and `<name>` is the name
+    /// of the resource itself.
+    const RESOURCE_NAME: &'static str;
```

Comment:
I thought that it will make sense to provide some kind of naming scheme that will also consider multiple implementations of the resource. I'll share a separate spec on that soon-ish, but the rough idea is:
Resource name is `common/object_store`, but what resource provider will work with is `common/object_store/gcp` or `common/object_store/in_memory` (or alternatively `state_keeper/io` -> `state_keeper/io/mempool_io` / `state_keeper/io/external_io`).

This will come in handy when the node configuration is defined by some kind of spec file (which is one of the main use cases for the framework).

Reply:
you can always keep a single resource type which is a wrapper for a dynamic implementation: `Box<dyn MyResourceTrait>` can also be a resource.

Reply:
Exactly, that's how [object store resource ](https://github.com/matter-labs/zksync-era/blob/776ec26e342993d419ce02ca80b743c82c01f98e/core/lib/node/src/resource/object_store.rs) is implemented